var/spack/repos/builtin/packages/py-nvidia-dali/package.py

```diff
@@ -168,4 +194,6 @@ class PyNvidiaDali(PythonPackage):
     depends_on("python@3.6:3.11", when="@1.23:", type=("build", "run"))
     depends_on("python@3.6:3.10", when="@:1.22", type=("build", "run"))
     depends_on("py-astunparse@1.6.0:", type=("build", "run"))
-    depends_on("py-gast@0.2.1:0.4.0", type=("build", "run"))
+    depends_on("py-gast@0.2.1:0.4.0", when="@:1.26", type=("build", "run"))
+    depends_on("py-gast@0.3.3:", when="@1.27:", type=("build", "run"))
+    depends_on("py-dm-tree", when="@1.27: ^python@3.7:", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("py-dm-tree", when="@1.27:", type=("build", "run"))
```
Spack only supports Python 3.7+ anyway